Index: docker-1.5/docker.c
===================================================================
--- docker-1.5.orig/docker.c
+++ docker-1.5/docker.c
@@ -79,7 +82,7 @@ void parse_cmd_line()
         g_printerr("-border requires a parameter\n");
         help = TRUE;
       }
-      } else if (0 == strcasecmp(argv[i], "-iconsize")) {
+    } else if (0 == strcasecmp(argv[i], "-iconsize")) {
       ++i;
       if (i < argc) {
         int s = atoi(argv[i]);
@@ -106,7 +118,7 @@ void parse_cmd_line()
       g_print("Usage: %s [OPTIONS]\n\n", argv[0]);
       g_print("Options:\n");
       g_print("  -help             Show this help.\n");
-      g_print("  -display DISLPAY  The X display to connect to.\n");
+      g_print("  -display DISPLAY  The X display to connect to.\n");
       g_print("  -border           The width of the border to put around the\n"
               "                    system tray icons. Defaults to 1.\n");
       g_print("  -vertical         Line up the icons vertically. Defaults to\n"
Index: docker-1.5/icons.h
===================================================================
--- docker-1.5.orig/icons.h
+++ docker-1.5/icons.h
@@ -4,6 +4,7 @@
 #include <glib.h>
 #include <X11/Xlib.h>
 #include "docker.h"
+#include "net.h"
 
 extern gboolean error;
 
